plugins/ksp/src/org/jetbrains/kotlin/ksp/KotlinSymbolProcessingPlugin.kt

```diff
@@ -56,8 +60,11 @@ class KotlinSymbolProcessingComponentRegistrar : ComponentRegistrar {
         val options = configuration[KSP_OPTIONS]?.apply {
             javaSourceRoots.addAll(contentRoots.filterIsInstance<JavaSourceRoot>().map { it.file })
         }?.build() ?: return
+        val messageCollector = configuration.get(CLIConfigurationKeys.MESSAGE_COLLECTOR_KEY)
+            ?: PrintingMessageCollector(System.err, MessageRenderer.PLAIN_FULL_PATHS, options.verboseMode)
```

Comment:
getNotNull is just throwing exception in case of null. In other scenarios, you need to supply a default message collector for the case of null and the default one is basically just swallowing all reports.

Reply:
My point is that since KSP always runs in `KotlinToJVMByteCodeCompiler`, which asserts non-null, you'll never get a null. On the other hand, if the environment doesn't provide a message collector, there's likely some problem. In that case it would be better to simply throw out an exception, instead of trying to patch it.

Reply:
> KSP always runs in KotlinToJVMByteCodeCompiler
This is only true for now, given we are going to MPP.

Reply:
I assume it will also be true for other platforms. If not, we better raise an exception rather than patch it and guess the behavior.